presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTime.java

```diff
@@ -17,32 +17,48 @@
 
 import java.time.Instant;
 import java.time.ZoneId;
+import java.time.ZoneOffset;
 import java.time.format.DateTimeFormatter;
 import java.util.Objects;
+import java.util.Optional;
 
 public final class SqlTime
 {
     private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss.SSS");
 
     private final long millisUtc;
-    private final TimeZoneKey sessionTimeZoneKey;
+    private final Optional<TimeZoneKey> sessionTimeZoneKey;
 
+    public SqlTime(long millisUtc)
+    {
+        this.millisUtc = millisUtc;
+        this.sessionTimeZoneKey = Optional.empty();
+    }
+
+    @Deprecated
```

Comment:
Instead of having two overloads of `SqlTime` (with one of them marked `@Deprecated`, what about having two factories: `SqlTime.of` and `SqlTime.ofLegacy`?

(Again, I prefer that. But this is not something I absolutely can't tolerate. I'll need to see what other people think.)